### DISCUSSION TITLE
Adding languages to docker containter

### DISCUSSION THREAD

**Discussion by 2-click** [microsoft/presidio#1059] [General]
Hi there, I am trying to add other languages to the presidio-analyzer. I did the following:
Added the desired language to [conf/default.yaml](https://github.com/microsoft/presidio/blob/main/presidio-analyzer/conf/default.yaml) as per https://microsoft.github.io/presidio/analyzer/languages/:
```
nlp_engine_name: spacy
models:
  -
    lang_code: en
    model_name: en_core_web_lg
  -
    lang_code: de
    model_name: de_core_news_lg
```

After that I ran
```
docker compose build
docker compose up -d
```
Note: during the build phase I can see it does something with the DE model, it's quite fast so I cannot exactly tell what, but it does something with it. 

Then I checked if the english analyzer works:
`curl -X POST http://localhost:5002/analyze -H "Content-type: application/json" --data "{ \"text\": \"John Smith drivers license is AC432223\", \"language\" : \"en\"}"`
That worked!

Then I tried german:
`curl -X POST http://localhost:5002/analyze -H "Content-type: application/json" --data "{ \"text\": \"Tim Burgmanns Rufnummer ist 054681571231\", \"language\" : \"de\"}"`

But it fails with the error:
`{"error":"No matching recognizers were found to serve the request."}`

I tried to investigate but I don't have any idea:
```
curl http://localhost:5002/recognizers
["UsBankRecognizer","UsItinRecognizer","UsLicenseRecognizer","UsSsnRecognizer","UsPassportRecognizer","NhsRecognizer","AuAcnRecognizer","CreditCardRecognizer","SgFinRecognizer","AuTfnRecognizer","AuAbnRecognizer","CryptoRecognizer","AuMedicareRecognizer","EmailRecognizer","DateRecognizer","IbanRecognizer","MedicalLicenseRecognizer","IpRecognizer","PhoneRecognizer","SpacyRecognizer","UrlRecognizer"]

```

I am missing something. Any ideas?

**Comment by omri374** [comment]
Hi, this seems like a duplicate of https://github.com/microsoft/presidio/issues/1058. The docker file spins up the flask in [app.py](https://github.com/microsoft/presidio/blob/main/presidio-analyzer/app.py#L40) which creates the default `AnalyzerEngine`. A simple solution would be to pass the `supported_languages` you are likely to have, and a more advanced solution would be to pass language codes when the app is created.

**Comment by 2-click** [comment]
Thanks a lot, for now that seems to work!
